apps/spark/src/main/java/com/linkedin/openhouse/jobs/scheduler/tasks/OperationTasksBuilder.java

```diff
@@ -37,6 +50,30 @@ private List<OperationTask<?>> prepareTableDirectoryOperationTaskList(
     return processMetadataList(directoryMetadataList, jobType);
   }
 
+  private List<OperationTask<?>> prepareDataLayoutOperationTaskList(JobConf.JobTypeEnum jobType) {
+    List<TableDataLayoutMetadata> tableDataLayoutMetadataList =
+        tablesClient.getTableDataLayoutMetadataList();
+    log.info("Fetched metadata for {} data layout strategies", tableDataLayoutMetadataList.size());
+    List<DataLayoutStrategy> strategies =
+        tableDataLayoutMetadataList.stream()
+            .map(TableDataLayoutMetadata::getDataLayoutStrategy)
+            .collect(Collectors.toList());
+    DataLayoutStrategyScorer scorer =
+        new SimpleWeightedSumDataLayoutStrategyScorer(
+            COMPACTION_GAIN_WEIGHT_DEFAULT, COMPUTE_COST_WEIGHT_DEFAULT);
+    List<ScoredDataLayoutStrategy> scoredStrategies = scorer.scoreDataLayoutStrategies(strategies);
+    DataLayoutCandidateSelector candidateSelector =
+        new GreedyMaxBudgetCandidateSelector(
+            MAX_COST_BUDGET_GB_HOURS_DEFAULT, MAX_STRATEGIES_COUNT_DEFAULT);
+    List<Integer> selectedStrategyIndices = candidateSelector.select(scoredStrategies);
+    log.info("Selected {} strategies", selectedStrategyIndices.size());
```

Comment:
Could we emit metrics here for selected strategies? For example, expected file count reduction and expected compute cost?

Reply:
Added